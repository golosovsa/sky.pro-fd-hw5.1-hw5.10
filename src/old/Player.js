import PlayerButton from './PlayerButton'
import TrackPlay from './TrackPlay'

function Player(props) {
  return (
    <div className="bar__player player">
      <div className="player__controls">
        <PlayerButton name="prev" action={props.action} />
        <PlayerButton name="play" action={props.action} />
        <PlayerButton name="next" action={props.action} />
        <PlayerButton name="repeat" action={props.action} />
        <PlayerButton name="shuffle" action={props.action} />
      </div>
      <TrackPlay
        author={props.author}
        album={props.album}
        action={props.action}
      />
    </div>
  )
}

export default Player
