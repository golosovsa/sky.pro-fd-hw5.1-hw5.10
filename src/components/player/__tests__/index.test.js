import ShallowRenderer from 'react-test-renderer/shallow'
import Player from '..'

describe('<Player />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot if track undefined', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<Player />)

      expect(tree).toMatchSnapshot()
    })

    it('should render successfully and match snapshot if track defined', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(
        <Player
          progress="progress"
          controlButtons="control buttons"
          likeButtons="like buttons"
          muteControl="mute control"
          track={{ name: 'name', author: 'author' }}
        />
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
