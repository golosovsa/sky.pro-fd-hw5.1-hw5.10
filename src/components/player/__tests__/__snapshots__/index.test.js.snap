// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> Snapshot test should render successfully and match snapshot if track defined 1`] = `
<styled.div>
  <styled.div />
  progress
  <PlayerPanel
    author="author"
    buttons="control buttons"
    likeButtons="like buttons"
    muteControl="mute control"
    name="name"
  />
</styled.div>
`;

exports[`<Player /> Snapshot test should render successfully and match snapshot if track undefined 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
